Assets/Scripts/Mixed/GameObjects/SharedMaterials.cs

```diff
@@ -25,20 +28,31 @@ public void Awake()
             InitializeDictionary();
         }
 
+        public void OnDestroy()
+        {
+            materialDictionary.Clear();
+            materialKeyByNameDictionary.Clear();
+            Instance = null;
+        }
+
         private void InitializeDictionary()
         {
             materialDictionary = new Dictionary<int, Material>();
+            materialKeyByNameDictionary = new Dictionary<string, int>();
+
+            var orderedMaterials = MaterialLibrary.SharedMaterials.OrderBy(m => m.Material.name).ToList();
 
             // Add materials to a dictionary to make querying faster.
-            foreach (var materialLib in MaterialLibrary.SharedMaterials)
-            {
-                materialDictionary.Add(materialLib.MaterialId, materialLib.Material);
+            for(var i = 0; i < orderedMaterials.Count; i++){
+                var material = orderedMaterials[i];
+                materialKeyByNameDictionary.Add(material.Material.name, i);
+                materialDictionary.Add(i, material.Material);
+                Debug.Log($"Added material: {material.Material.name} to library.");
```

Comment:
It would be great to generate a warning or console log if there are duplicates names